android/core/src/test/java/com/pega/vehicledamagemodeling/api/SelectionServiceTest.java

```diff
@@ -0,0 +1,90 @@
+package com.pega.vehicledamagemodeling.api;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonObject;
+
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+
+public class SelectionServiceTest {
+
+    private static String Selection = "selection";
+    private static String Id = "id";
+    private static String MainScreenText = "mainScreenText";
+
+    @Test
+    public void whenJsonContainsTwoPartsThenReturnCorrectInitJson() {
+        //given
+        JsonObject expectedJson = new JsonObject();
+        JsonArray partsArray = new JsonArray();
+        JsonObject jsonProperty = new JsonObject();
+        jsonProperty.addProperty(Id,"roof");
+        partsArray.add(jsonProperty);
+        JsonObject jsonProperty2 = new JsonObject();
+        jsonProperty2.addProperty(Id,"front bumper");
+        partsArray.add(jsonProperty2);
+        expectedJson.add(Selection,partsArray);
+
+        SelectionService selectionService = new SelectionService();
+        selectionService.attachJson(expectedJson);
+
+        //when
+        JsonObject result = selectionService.getInitJson();
+
+        //then
+        assertEquals(expectedJson,result);
+    }
+
+    @Test
+    public void whenJsonIsEmptyThenReturnEmptyInitJson(){
+        //given
+        JsonObject initJson = new JsonObject();
+        SelectionService selectionService = new SelectionService();
+        selectionService.attachJson(initJson);
+
+        //when
+        JsonObject result = selectionService.getInitJson();
+
+        //then
+        assertEquals(initJson,result);
+    }
+
+    @Test
+    public void whenInitJsonContainTextThenReturnCorrectText() {
+        //given
+        JsonObject initJson = new JsonObject();
+        initJson.addProperty(MainScreenText, "nothing");
+        SelectionService selectionService = new SelectionService();
```

Comment:
Those tests aren't unit tests... This sounds like integration tests. They test 2-3 classes, not one. You can check by changing the implementation of Parser, those tests will fail